### PR TITLE
quickder: 1.3.0 -> 1.7.1

### PR DESCRIPTION
**Pull request by leungbk**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fufexan** [comment]
Result of `nixpkgs-review pr 184588` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>arpa2common</li>
    <li>quickder</li>
    <li>quickmem</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>arpa2cm</li>
  </ul>
</details>

**Comment by fufexan** [comment]
Nix reports `sha256-UpAVyDXCe07ZwjD307t6G9f/Nny4QYXxGxft1KsiYYg=` as the correct `arpa2common` source hash.

**Comment by fufexan** [comment]
Result of `nixpkgs-review pr 184588` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>arpa2cm</li>
    <li>arpa2common</li>
    <li>quickder</li>
    <li>quickmem</li>
  </ul>
</details>

**Review by fufexan — APPROVED** [review]
LGTM

File: pkgs/development/libraries/quickder/default.nix [hunk 1]
```diff
@@ -1,50 +1,58 @@
...
 
-  src = fetchFromGitHub {
-    sha256 = "15lxv8vcjnsjxg7ywcac5p6mj5vf5pxq1219yap653ci4f1liqfr";
-    rev = "version-${version}";
-    owner = "vanrein";
+  src = fetchFromGitLab {
+    sha256 = "sha256-f+ph5PL+uWRkswpOLDwZFWjh938wxoJ6xocJZ2WZLEk=";
+    rev = "v${version}";
+    owner = "arpa2";
     repo = "quick-der";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/quickder/default.nix:22]:
```suggestion
    owner = "arpa2";
    repo = "quick-der";
    rev = "v${version}";
    sha256 = "sha256-f+ph5PL+uWRkswpOLDwZFWjh938wxoJ6xocJZ2WZLEk=";
```

File: pkgs/development/libraries/quickder/default.nix [hunk 2]
```diff
@@ -1,50 +1,58 @@
...
-  postPatch = ''
-    substituteInPlace ./CMakeLists.txt \
-      --replace "get_version_from_git" "set (Quick-DER_VERSION 1.2) #"
-    substituteInPlace ./CMakeLists.txt \
-      --replace \$\{ARPA2CM_TOOLCHAIN_DIR} "$out/share/ARPA2CM/toolchain/"
-    patchShebangs python/scripts/
-  '';
+  buildInputs =
+    [
+      arpa2cm
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/quickder/default.nix:34]:
```suggestion
  buildInputs = [
    arpa2cm
```
and so on

File: pkgs/development/libraries/quickmem/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "quickmem";
+  version = "0.3.0";
+
+  src = fetchFromGitLab {
+    sha256 = "sha256-cqg8QN4/I+zql7lVDDAgFA05Dmg4ylBTvPSPP7WATdc=";
+    rev = "v${finalAttrs.version}";
+    owner = "arpa2";
+    repo = "Quick-MEM";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/quickmem/default.nix:19]:
```suggestion
    owner = "arpa2";
    repo = "Quick-MEM";
    rev = "v${version}";
    sha256 = "sha256-cqg8QN4/I+zql7lVDDAgFA05Dmg4ylBTvPSPP7WATdc=";
```

File: pkgs/development/libraries/quickmem/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+, stdenv
+, fetchFromGitLab
+, cmake
+, doxygen
+, graphviz
+, arpa2common
+, arpa2cm
+}:
+
+stdenv.mkDerivation (finalAttrs: {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/quickmem/default.nix:11]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/libraries/quickmem/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    cmake
+    doxygen
+    graphviz
+  ];
+
+  buildInputs =
+    [
+      arpa2cm
+      arpa2common
+    ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/quickmem/default.nix:32]:
```suggestion
  buildInputs = [
    arpa2cm
    arpa2common
  ];
```

File: pkgs/development/libraries/quickmem/default.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "Memory pooling for ARPA2 projects";
+    homepage = "https://gitlab.com/arpa2/Quick-MEM/";
+    license = licenses.bsd2;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/quickmem/default.nix:41]: This is not allowed. maintainers must be set presumable to yourself.

File: pkgs/development/tools/build-managers/arpa2cm/default.nix [hunk 7]
```diff
@@ -2,10 +2,10 @@
...
   pname = "arpa2cm";
-  version = "0.9.0";
+  version = "1.0.4";
 
   src = fetchFromGitLab {
-    sha256 = "sha256-1z0fH8vZJiPkY/C654us9s2BULM1tlvvYcszNqk34yI=";
+    sha256 = "sha256-2vb/7UL+uWGrQNh8yOZ3gih5G1/eOp064hF78SDsPGk=";
     rev = "v${version}";
     repo = pname;
     owner = "arpa2";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/arpa2cm/default.nix:11]:
```suggestion
    owner = "arpa2";
    repo = pname;
    rev = "v${version}";
    sha256 = "sha256-2vb/7UL+uWGrQNh8yOZ3gih5G1/eOp064hF78SDsPGk=";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by fufexan — APPROVED** [review]

**Pull request merged**